### PR TITLE
correctly set spree_option_value_product_personalization_id

### PR DESCRIPTION
**Pull request by threadhead**
Will add additional specs. DO NOT MERGE

### CODE REVIEW

File: spec/models/spree/line_item_personalization_spec.rb [hunk 1]
```diff
@@ -10,10 +10,16 @@
     @product = @variant.product
     @product_personalizations  = @variant.product.personalizations
 
+    variant2 = create(:variant_with_personalizations, price: 67)
+    ppwov = create(:product_personalization_with_option_value)
+    ppwov.product = variant2.product
+    ppwov.save
```
Comment (mtuckergd) [inline spec/models/spree/line_item_personalization_spec.rb]: this should probably be `save!` to ensure any issue doesn't fail silently

File: spec/models/spree/line_item_personalization_spec.rb [hunk 2]
```diff
@@ -10,10 +10,16 @@
...
+
     @product_personalization_with_option_value = create(:product_personalization_with_option_value)
     @product_personalization_with_option_value.product = @product
     @product_personalization_with_option_value.save
 
+
```
Comment (mtuckergd) [inline spec/models/spree/line_item_personalization_spec.rb]: why the extra line break?

File: app/models/spree/line_item_personalization.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@ class LineItemPersonalization < ActiveRecord::Base
     validate :value_length
 
     before_validation { self.value = self.value.try(:strip) }
+    before_save :set_ovpp_id
```
Comment (mtuckergd) [inline app/models/spree/line_item_personalization.rb]: Is there any change that the `spree_option_value_product_personalization_id` ever changes for a line_item_personalization object? If not, then it seems pretty inefficient to keep setting it on every save, vs only the first time. I'm just not clear on whether that value could ever change.

Comment (IrvinFan) [inline app/models/spree/line_item_personalization.rb]: I agree. The option_value_id should be set in `copy_personalizations` (similar to how we set `spree_product_personalization_id`), which is only invoked for `:create`

I'll make the change.

**Comment by mtuckergd** [comment]
👍  for me. Will wait for @threadhead to comment on the changes before we merge.

**Comment by IrvinFan** [comment]
LGTM 👍

**Pull request merged**